### PR TITLE
some improvements to handling of the NIRS optode definition

### PR DESCRIPTION
**Pull request by robertoostenveld**
see http://bugzilla.fieldtriptoolbox.org/buglist.cgi?quicksearch=nirs&list_id=5399 and http://bugzilla.fieldtriptoolbox.org/show_bug.cgi?id=3027 (although this is not specifically related to Homer)

### CODE REVIEW

**Comment by robertoostenveld** [comment]
@Horschig could you have a quick look at the files that I changed for this? I used the documentation from http://www.fieldtriptoolbox.org/development/nirs.

regarding actual support in data processing functions, I only did two, since I realize that it is a bit difficult not being able to test. Most changes are in support functions.

**Comment by robertoostenveld** [comment]
rather than linking it to 3027, http://bugzilla.fieldtriptoolbox.org/show_bug.cgi?id=2753 would have been a better match.

File: forward/private/ft_datatype_sens.m [hunk 1]
```diff
@@ -22,26 +26,33 @@
...
+%    sens.chanpos  = Mx3 matrix with channel positions (often the same as electrode positions)
+%    sens.tra      = MxN matrix to combine electrodes into channels
 % In case sens.tra is not present in the EEG sensor array, the channels
 % are assumed to be average referenced.
 %
+% The structure for NIRS channels contains
+%   sens.optopos        = contains information about the position of the optodes
+%   sens.optotype       = contains information about the type of optode (receiver or transmitter)
+%   sens.chanpos        = contains information about the position of the channels (i.e. average of optopos)
+%   sens.tra            = NxC matrix, boolean, contains information about how receiver and transmitter form channels
```
Comment (Horschig) [inline forward/private/ft_datatype_sens.m]: This is actually called sens.transceiver. We decided that .tra is too ambiguous

File: private/ft_fetch_sens.m [hunk 2]
```diff
@@ -113,66 +132,33 @@
...
-    sens.unit = dum.unit;
-  end
-  if isfield(dum,'coordsys')
-    sens.coordsys = dum.coordsys;
-  end
+  sens = keepfields(sens, {'elecpos', 'chanpos', 'unit', 'coordsys', 'label'});
+  
+elseif hasoptofile
+  display('reading optodes from file ''%s''\n', cfg.optofile);
+  sens = ft_read_sens(cfg.elecfile);
```
Comment (Horschig) [inline private/ft_fetch_sens.m]: typo here, should be cfg.optofile

File: private/ft_fetch_sens.m [hunk 3]
```diff
@@ -113,66 +132,33 @@
...
+elseif hascfgopto
+  display('using optodes specified in the configuration\n');
+  sens = cfg.opto;
```
Comment (Horschig) [inline private/ft_fetch_sens.m]: should this be added? 
sens = keepfields(sens, {'optopos', 'optotype', 'chanpos', 'unit', 'coordsys', 'label', 'transceiver', 'wavelength', 'transmits', 'laserstrength'});

File: private/ft_fetch_sens.m (hunk 2 above)
Comment (robertoostenveld) [inline private/ft_fetch_sens.m]: [bug3027-nirs c3b7e30] FIX - made some changes as suggested by Jorn as comments to the commit, thanks!
 1 file changed, 7 insertions(+), 1 deletion(-)

File: private/ft_fetch_sens.m (hunk 3 above)
Comment (robertoostenveld) [inline private/ft_fetch_sens.m]: [bug3027-nirs c3b7e30] FIX - made some changes as suggested by Jorn as comments to the commit, thanks!
 1 file changed, 7 insertions(+), 1 deletion(-)

**Comment by robertoostenveld** [comment]
I'll commit this for now. Hopefully we can continue work on NIRS in the near future.

**Pull request merged**